packages/white-whale-std/src/fee.rs

```diff
@@ -62,11 +62,125 @@ impl VaultFee {
     }
 }
 
+/// Represents the fee structure for transactions within a pool.
+///
+///
+/// # Fields
+/// - `protocol_fee`: The fee percentage charged by the protocol on each transaction to support
+///   operational and developmental needs.
+/// - `swap_fee`: The fee percentage allocated to liquidity providers as a reward for supplying
+///   liquidity to the pool, incentivizing participation and ensuring pool health.
+/// - `burn_fee`: A fee percentage that is burned on each transaction, helping manage the token
+///   economy by reducing supply over time, potentially increasing token value.
+/// - `osmosis_fee` (optional): Specific to the Osmosis feature, this fee is charged on each
+///   transaction when the Osmosis feature is enabled, supporting specific ecosystem requirements.
+/// - `extra_fees`: A vector of custom fees allowing for extensible and adaptable fee structures
+///   to meet diverse and evolving needs. Validation ensures that the total of all fees does not
+///   exceed 100%, maintaining fairness and avoiding overcharging.
+///
+/// # Features
+/// - `osmosis`: Enables the `osmosis_fee` field, integrating specific fee requirements for the
+///   Osmosis protocol within the pool's fee structure.
+#[cw_serde]
+pub struct PoolFee {
+    /// Fee percentage charged on each transaction for the protocol's benefit.
+    pub protocol_fee: Fee,
+
+    /// Fee percentage allocated to liquidity providers on each swap.
+    pub swap_fee: Fee,
+
+    /// Fee percentage that is burned on each transaction. Burning a portion of the transaction fee
+    /// helps in reducing the overall token supply.
+    pub burn_fee: Fee,
+
+    /// Fee percentage charged on each transaction specifically for Osmosis integrations. This fee
+    /// is only applicable when the `osmosis` feature is enabled
+    #[cfg(feature = "osmosis")]
+    pub osmosis_fee: Fee,
+
+    /// A list of custom, additional fees that can be defined for specific use cases or additional
+    /// functionalities. This vector enables the flexibility to introduce new fees without altering
+    /// the core fee structure. Total of all fees, including custom ones, is validated to not exceed
+    /// 100%, ensuring a balanced and fair fee distribution.
+    pub extra_fees: Vec<Fee>,
```

Comment:
👀

Reply:
Is this something you came up with or did you discuss it with @Sen-Com ? I like the idea of having a vector of fees so we don't need to be changing the core structure constantly 👍

Reply:
Yeah that sounds great, good idea